central/networkgraph/service/service_impl_postgres_test.go

```diff
@@ -0,0 +1,517 @@
+		Type: storage.NetworkEntityInfo_DEPLOYMENT,
+		Id:   deployment.Id,
+		Desc: &storage.NetworkEntityInfo_Deployment_{
+			Deployment: &storage.NetworkEntityInfo_Deployment{
+				Namespace: deployment.Namespace,
+			},
+		},
+	}
+
+	entityInfo := entity.GetInfo()
+
+	if ingress {
+		return &storage.NetworkFlow{
+			Props: &storage.NetworkFlowProperties{
+				SrcEntity:  entityInfo,
+				DstEntity:  deploymentEntityInfo,
+				DstPort:    1234,
+				L4Protocol: storage.L4Protocol_L4_PROTOCOL_TCP,
+			},
+			LastSeenTimestamp: nil,
+			ClusterId:         deployment.ClusterId,
+		}
+	}
+	return &storage.NetworkFlow{
+		Props: &storage.NetworkFlowProperties{
+			DstEntity:  entityInfo,
+			SrcEntity:  deploymentEntityInfo,
+			DstPort:    1234,
+			L4Protocol: storage.L4Protocol_L4_PROTOCOL_TCP,
+		},
+		LastSeenTimestamp: nil,
+		ClusterId:         deployment.ClusterId,
+	}
+}
+
+func (s *networkGraphServiceSuite) TestGetExternalNetworkFlows() {
+	ctx := sac.WithGlobalAccessScopeChecker(
+		context.Background(),
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
+			sac.ResourceScopeKeys(resources.Deployment, resources.NetworkGraph),
+		),
+	)
+
+	globalWriteAccessCtx := sac.WithGlobalAccessScopeChecker(context.Background(),
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS, storage.Access_READ_WRITE_ACCESS),
+			sac.ResourceScopeKeys(resources.NetworkGraph, resources.Deployment)))
+
+	entityID, _ := externalsrcs.NewClusterScopedID(testCluster, "192.168.1.1/32")
+	entityID2, _ := externalsrcs.NewClusterScopedID(testCluster, "10.0.0.2/32")
+	entityID3, _ := externalsrcs.NewClusterScopedID(testCluster, "1.1.1.1/32")
+
+	entity := testutils.GetExtSrcNetworkEntity(entityID.String(), "ext1", "192.168.1.1/32", false, testCluster)
+	entity2 := testutils.GetExtSrcNetworkEntity(entityID2.String(), "ext2", "10.0.0.2/32", false, testCluster)
+	entity3 := testutils.GetExtSrcNetworkEntity(entityID3.String(), "ext3", "10.0.100.25/32", false, testCluster)
+
+	entities := []*storage.NetworkEntity{
+		entity,
+		entity2,
+		entity3,
+	}
+
+	deployment := &storage.Deployment{
+		Id:        fixtureconsts.Deployment1,
+		ClusterId: testCluster,
+		Namespace: testNamespace,
+	}
+
+	deployment2 := &storage.Deployment{
+		Id:        fixtureconsts.Deployment2,
+		ClusterId: testCluster,
+		Namespace: testNamespace,
+	}
+
+	err := s.deploymentsDataStore.UpsertDeployment(globalWriteAccessCtx, deployment)
+	s.NoError(err)
+
+	err = s.deploymentsDataStore.UpsertDeployment(globalWriteAccessCtx, deployment2)
+	s.NoError(err)
+
+	for _, e := range entities {
+		err = s.entityDataStore.CreateExternalNetworkEntity(globalWriteAccessCtx, e, true)
+		s.NoError(err)
+	}
+
+	deploymentToEntity1 := externalFlow(deployment, entity, false)
+	deploymentToEntity3 := externalFlow(deployment, entity3, false)
+
+	deployment2ToEntity3 := externalFlow(deployment2, entity3, false)
+
+	// flow from deployment to es1a and es1c but not es1b
```

Comment:
What does this mean? What are es1a/b/c?

Reply:
This is left over from an early version - apologies, I'll fix when I update the naming